### PR TITLE
pipenv-2018.11.26

### PR DESCRIPTION
**Pull request by roccqqck**
Checklist

- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vend other packages
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/pipenv/LICENSE [hunk 1]
```diff
@@ -0,0 +1,22 @@
+The MIT License (MIT)
```
Comment (chrisburr) [inline recipes/pipenv/LICENSE]: This file isn't needed, `conda-build` will find it in the source tarball.

(The `licence_file` entry is still needed)

File: recipes/pipenv/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  entry_points:
+    - pipenv=pipenv:cli
+    - pipenv-resolver=pipenv.resolver:main
+  script: 
+    - python -m pip install --no-deps --ignore-installed ./pipenv-2018.11.26-py3-none-any.whl
+
+requirements:
+  host:
+    - certifi
+    - pip >=9.0.1
```
Comment (chrisburr) [inline recipes/pipenv/meta.yaml]: You probably only need python and pip here.

File: recipes/pipenv/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+{% set version = "2018.11.26" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+
+# cuz source code didn't work, i used wheel instead.
+source:
+  url: "https://files.pythonhosted.org/packages/13/b4/3ffa55f77161cff9a5220f162670f7c5eb00df52e00939e203f601b0f579/pipenv-2018.11.26-py3-none-any.whl"
```
Comment (chrisburr) [inline recipes/pipenv/meta.yaml]: All conda-forge packages should be build from source:
```suggestion
  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```

File: recipes/pipenv/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  sha256: "56ad5f5cb48f1e58878e14525a6e3129d4306049cb76d2f6a3e95df0d5fc6330"
+
+build:
+  noarch: python
+  number: 0
+  entry_points:
+    - pipenv=pipenv:cli
+    - pipenv-resolver=pipenv.resolver:main
+  script: 
+    - python -m pip install --no-deps --ignore-installed ./pipenv-2018.11.26-py3-none-any.whl
```
Comment (chrisburr) [inline recipes/pipenv/meta.yaml]:
```suggestion
    - python -m pip install . -vv
```

File: recipes/pipenv/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+
+# cuz source code didn't work, i used wheel instead.
+source:
+  url: "https://files.pythonhosted.org/packages/13/b4/3ffa55f77161cff9a5220f162670f7c5eb00df52e00939e203f601b0f579/pipenv-2018.11.26-py3-none-any.whl"
+  sha256: "56ad5f5cb48f1e58878e14525a6e3129d4306049cb76d2f6a3e95df0d5fc6330"
```
Comment (chrisburr) [inline recipes/pipenv/meta.yaml]: Quotes on the hash prevent the update bot from working
```suggestion
  sha256: 56ad5f5cb48f1e58878e14525a6e3129d4306049cb76d2f6a3e95df0d5fc6330
```

File: recipes/pipenv/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    - virtualenv-clone >=0.2.5
+
+test:
+  imports:
+    - pipenv
+  commands:
+    - pipenv --help
+
+about:
+  home: https://pipenv-fork.readthedocs.io
```
Comment (chrisburr) [inline recipes/pipenv/meta.yaml]: Is this the right URL? I think it should be GitHub (to match with PyPI)
```suggestion
  home: https://github.com/pypa/pipenv
```
or:
```suggestion
  home: https://pipenv.kennethreitz.org
```

File: recipes/pipenv/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: "Python Development Workflow for Humans."
+  dev_url: https://github.com/pypa/pipenv
```
Comment (chrisburr) [inline recipes/pipenv/meta.yaml]:
```suggestion
  dev_url: https://github.com/pypa/pipenv
  doc_url: https://pipenv.kennethreitz.org
```

File: recipes/pipenv/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,56 @@
+  
+{% set name = "pipenv" %} 
+{% set version = "2018.11.26" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+
+# cuz source code didn't work, i used wheel instead.
```
Comment (chrisburr) [inline recipes/pipenv/meta.yaml]:
```suggestion
```

File: recipes/pipenv/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,56 @@
+
```
Comment (chrisburr) [inline recipes/pipenv/meta.yaml]:
```suggestion
```

File: recipes/pipenv/meta.yaml (hunk 8 above)
Comment (roccqqck) [reply recipes/pipenv/meta.yaml]: i tried pipenv source to build before
it didnt work, it always showed error, so i change it to wheel
i will try source again

Comment (roccqqck) [reply recipes/pipenv/meta.yaml]: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=80344
it shows errors again now
i still dont know why
i think there are some bug in the source code

Comment (chrisburr) [reply recipes/pipenv/meta.yaml]: If you scroll up to the first error it tells you there are missing dependencies:
```
    RuntimeError: Setuptools downloading is disabled in conda build. Be sure to add all dependencies in the meta.yaml  url=https://pypi.org/simple/parver/
```
The solution is to add `parver` to the `host` dependencies.

Comment (chrisburr) [reply recipes/pipenv/meta.yaml]: > please let me use wheel

It's very rare for wheels to be accepted for the reasons outlined [here](https://github.com/conda-forge/tensorflow-probability-feedstock/issues/3#issuecomment-462491626). Exceptions are extremely rare and only made after it's been shown building from source is too difficult. (tensorflow is now being built from source so even that exception no longer applies)

Comment (roccqqck) [reply recipes/pipenv/meta.yaml]: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=80348

it shows errors again
i cant figure why

Comment (roccqqck) [reply recipes/pipenv/meta.yaml]: > > please let me use wheel
> 
> It's very rare for wheels to be accepted for the reasons outlined [here](https://github.com/conda-forge/tensorflow-probability-feedstock/issues/3#issuecomment-462491626). Exceptions are extremely rare and only made after it's been shown building from source is too difficult. (tensorflow is now being built from source so even that exception no longer applies)

i built pipenv in my channel before
pipenv-2018.5.18 can be built from source
after that i had to build from wheel
the dependencies are the same

i just dont know why

**Pull request closed without merging**